sci-geosciences/qgis/qgis-3.0.1-r1.ebuild

```diff
@@ -0,0 +1,224 @@
+	)
+	webkit? ( >=dev-qt/qtwebkit-${QT_MIN_VER}:5 )
+"
+DEPEND="${COMMON_DEPEND}
+	>=dev-qt/linguist-tools-${QT_MIN_VER}:5
+	>=dev-qt/qttest-${QT_MIN_VER}:5
+	>=dev-qt/qtxmlpatterns-${QT_MIN_VER}:5
+	sys-devel/bison
+	sys-devel/flex
+"
+RDEPEND="${COMMON_DEPEND}
+	sci-geosciences/gpsbabel
+"
+
+# Disabling test suite because upstream disallow running from install path
+RESTRICT="test"
+
+PATCHES=(
+	# git master
+	"${FILESDIR}/${PN}-2.18.12-cmake-lib-suffix.patch"
+	# release-3_0 branch
+	"${FILESDIR}/${P}-check-constraints.patch"
+	# TODO upstream
+	"${FILESDIR}/${P}-qt-5.11.patch"
+	"${FILESDIR}/${PN}-3.0.0-featuresummary.patch"
+)
+
+pkg_setup() {
+	use python && python-single-r1_pkg_setup
+}
+
+src_prepare() {
+	cmake-utils_src_prepare
+
+	sed -e "s:\${QT_BINARY_DIR}:$(qt5_get_bindir):" \
+		-i CMakeLists.txt || die "Failed to fix lrelease path"
+
+	sed -e "/QT_LRELEASE_EXECUTABLE/d" \
+		-e "/QT_LUPDATE_EXECUTABLE/s/set/find_program/" \
+		-e "s:lupdate-qt5:NAMES lupdate PATHS $(qt5_get_bindir) NO_DEFAULT_PATH:" \
+		-i cmake/modules/ECMQt4To5Porting.cmake || die "Failed to fix ECMQt4To5Porting.cmake"
+
+	cd src/plugins || die
+}
+
+src_configure() {
+	local mycmakeargs=(
+		-DQGIS_MANUAL_SUBDIR=/share/man/
+		-DBUILD_SHARED_LIBS=ON
+		-DQGIS_LIB_SUBDIR=$(get_libdir)
+		-DQGIS_PLUGIN_SUBDIR=$(get_libdir)/qgis
+		-DQWT_INCLUDE_DIR=/usr/include/qwt6
+		-DQWT_LIBRARY=/usr/$(get_libdir)/libqwt6-qt5.so
+		-DPEDANTIC=OFF
+		-DWITH_APIDOC=OFF
+		-DWITH_QSPATIALITE=ON
+		-DENABLE_TESTS=OFF
+		-DWITH_3D=$(usex 3d)
+		-DWITH_GEOREFERENCER=$(usex georeferencer)
+		-DWITH_GRASS=$(usex grass)
+		-DWITH_SERVER=$(usex mapserver)
+		-DWITH_ORACLE=$(usex oracle)
+		-DWITH_QWTPOLAR=$(usex polar)
+		-DWITH_POSTGRESQL=$(usex postgres)
+		-DWITH_BINDINGS=$(usex python)
+		-DWITH_CUSTOM_WIDGETS=$(usex python)
+		-DWITH_QTWEBKIT=$(usex webkit)
+	)
+
+	if use grass; then
+		mycmakeargs+=(
+			-DWITH_GRASS7=ON
+			-DGRASS_PREFIX7=/usr/$(get_libdir)/grass70
+		)
+	fi
+
+	use python && mycmakeargs+=( -DBINDINGS_GLOBAL_INSTALL=ON )
+
+	# bugs 612956, 648726
+	addpredict /dev/dri/renderD128
+	addpredict /dev/dri/renderD129
+
+	cmake-utils_src_configure
+}
+
+src_install() {
+	cmake-utils_src_install
+
+	domenu debian/qgis.desktop
+
+	local size type
+	for size in 16 22 24 32 48 64 96 128 256; do
+		newicon -s ${size} debian/icons/${PN}-icon${size}x${size}.png ${PN}.png
+		newicon -c mimetypes -s ${size} debian/icons/${PN}-mime-icon${size}x${size}.png ${PN}-mime.png
+		for type in qgs qml qlr qpt; do
+			newicon -c mimetypes -s ${size} debian/icons/${PN}-${type}${size}x${size}.png ${PN}-${type}.png
+		done
+	done
+	newicon -s scalable images/icons/qgis_icon.svg qgis.svg
+
+	insinto /usr/share/mime/packages
+	doins debian/qgis.xml
+
+	if use examples; then
+		docinto examples
+		dodoc -r "${WORKDIR}"/qgis_sample_data/.
+		docompress -x /usr/share/doc/${PF}/examples
+	fi
+
+	if use python; then
+		python_optimize "${ED%/}"/usr/share/qgis/python
+
+		if use grass; then
```

Comment:
Since this is now behind the `use python` conditional, you have to make `grass` require `python` in `REQUIRED_USE`. Otherwise it would be possible to build with USE="grass -python" but then miss that python_fix_shebang.

Reply:
I see... Yep I missed that use case.
Indeed, based on this code, grass does indeed need python...
Which will actually be taken care implicitly since sci-geosciences/grass-7.2.0-r1 depends on python2_7 anyway. I agree it is better to be explicit here, so I changed the ebuild and am testing now...